sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -102,12 +180,12 @@ def cat_split(
             for t in tensors
         ]
     )
-
+    # TODO: Need to transfer tensors to same devices
     shard_dim = tensors[0].shard_dim
     shard_count = tensors[0].shard_count
     if dim != shard_dim:
         shards = [cat(shards, dim) for shards in zip(*[t.shards for t in tensors])]
-        return SplitPrimitiveTensor(ts=shards, shard_dim=shard_dim)
+        return SplitPrimitiveTensor(ts=shards, shard_dim=shard_dim,devices=tensors[0].devices, pinned=False)
```

Comment:
We can remove all of these sets now right? We want the wrapper to set the right device information above.

Reply:
We can drop the pinned and let the wrapper set it to whatever the input value was. But devices we can't. The wrapper only transfers the input shards and sets devices on the input, it won't set it on the output. If we don't have it in here, then the `SplitPrimitiveTensor` will default to `devices=tuple(range(len(shard_count))` rather than the location of the pinned tensor in the input.

Although now that I look at cat longer, it should always through an error on the input since we're unlike to ever cat 1 pinned tensor with N unpinned tensors. Would it not be used primarily to cat unpinned tensors?